### DISCUSSION TITLE
FileSystem API directory compression method

### DISCUSSION THREAD

**Discussion by ktilelis** [gildas-lormeau/zip.js#224] [Q&A] [answered]
Hello! 

I'm using the FileSystem API (version 2.2.17 of jszip) to create a zip file that contains both files and folders. The folders can contain files. I add the directory using the addDirectory method and the files using the addText method.

When examining the zip file using 7zip I see that the method for files is DEFLATE while for directories is STORED. 
Is it possible to compress the directories so that the method is DEFLATE instead of STORE? How can I specify that through the FileSystem API?

BR,
Kostas

**Comment by gildas-lormeau** [comment, marked as answer]
Hi,

You cannot change the method because the directory entries aren't present in the zip file actually. You can see this by listing the entries in the zip file with a program like  `unzip`.

**Reply by ktilelis** [reply]
Thank you for your prompt reply and for creating such a useful library.

**Reply by gildas-lormeau** [reply]
You're welcome!
